diesel/src/main.rs

```diff
@@ -172,8 +170,8 @@ fn main() {
                     //  below).
                     //
                     // This does not work
-                    .with_config(add2, |cfg| {
-                        cfg.0.limit(4096); // <- limit size of the payload
+                    .with_async_config(add2, |(json_cfg, )| {
```

Comment:
For methods that return future as response, you need `with_async*`, not plain `with`